go/store/nbs/generational_chunk_store.go

```diff
@@ -87,11 +112,29 @@ func (gcs *GenerationalNBS) GetMany(ctx context.Context, hashes hash.HashSet, fo
 		return err
 	}
 
-	if len(notInOldGen) == 0 {
+	if len(notFound) == 0 {
 		return nil
 	}
 
-	return gcs.newGen.GetMany(ctx, notInOldGen, found)
+	err = gcs.newGen.GetMany(ctx, notFound, func(ctx context.Context, chunk *chunks.Chunk) {
+		func() {
+			mu.Lock()
+			defer mu.Unlock()
+			delete(notFound, chunk.Hash())
+		}()
+
+		found(ctx, chunk)
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// We've been asked for objets which we don't apparently have. Check the lost gen commits that we know about.
```

Comment:
Seemingly stale comment. At this point we didn't even check `len(notFound) != 0`.

Reply:
not checking len(notFound) was a bug. Fixed.